keps/sig-windows/1981-windows-privileged-container-support/README.md

```diff
@@ -472,41 +472,62 @@ Because Windows privileged containers will work much differently than Linux priv
-- Client libraries such as https://pkg.go.dev/k8s.io/client-go/rest#InClusterConfig may be updated to prefix paths with `$CONTAINER_SANDBOX_MOUNT_POINT` if the environment variable is set for Windows so these libraries will work in `hostProcess`containers. This will be re-evaluated when transistioning from `alpha` to `beta` as we get more feedback.
-Note: it is not possible to feature-gate this behavior in client libraries and because of this the functionality should not be added to client libraries after privileged containers while this feature is in `alpha`.
-- Named Pipe mounts will **not** be supported. Instead named pipes should access via their path on the host (\\\\.\\pipe\\*). Unix domain sockets mounts **will** be supported.
-- All other volume types supported for normal containers on Windows will work with privileged containers.
+  - Note: We are prototyping a new approach to how the file system is created for `hostProcess` containers that would present the filesystem in a similar manner to non-hostProcess containers running on Windows (`c:\` (trailing \ included) would be the root instead of `c:\c\<container id>\`).
+  This would make it so files from volume mounts would be accessible via static paths. HostProcess containers would still have full access to the host file-system.
+  https://github.com/microsoft/hcsshim/pull/1107 is tracking this exploratory work.
+  This functionality will most-likely not be ready during Kubernetes v1.23 and any changes made to how volume mounts work would be done before this features becomes stable.
+
+- Client libraries such as https://pkg.go.dev/k8s.io/client-go/rest#InClusterConfig may be updated to prefix paths with `$CONTAINER_SANDBOX_MOUNT_POINT` if the environment variable is set for Windows so these libraries will work in `hostProcess` containers. This will be re-evaluated when transitioning from `alpha` to `beta` as we get more feedback.
+  - Note: it is not possible to feature-gate this behavior in client libraries and because of this the functionality should not be added to client libraries after `hostProcess` containers while this feature is in `alpha`.
+  - [kubernetes/kubernetes#104490](https://github.com/kubernetes/kubernetes/pull/104490) add support for `HostProcess` containers to the golang client library.
+- Named Pipe mounts will **not** be supported. Instead named pipes should be accessed via their path on the host (\\\\.\\pipe\\*).
+- Unix domain sockets mounts support will be added before `HostProcess` containers graduate to `stable`. For `alpha` and `beta` Unix domain sockets can be accessed via their paths on the host like named pipes.
```

Comment:
This is fairly important for CSI node plugins. The daemonset deployment [YAMLs](https://github.com/kubernetes-csi/csi-proxy#setup-for-csi-driver-deployment) in CSI plugins and the CSI Node Driver Registrar for Windows typically specify multiple unix domain socket file paths - the location of which, so far, have been surfaced using volume mounts.

Reply:
Do these plugins look in hardcoded paths (e.g are they expected to be surfaced at the same spot in every container)?

Reply:
It's surfaced today using hostpath but since hostprocess containers get access to the entire root filesystem, is it required?

Reply:
What I'm trying to figure out is, do these plugins expect there to be a uds at "C:\\path\\to\\uds.sock" or similar in every container? Sure, the container could just access the socket on the host here but if the plugin itself is expecting some hard coded path, as it can reasonably make that assumption in a normal container scenario that has more isolation guarantees, then that won't work here currently. Basically are there any code changes needed for the plugins, or do they all expose some flag to pass a custom uds path and have it just point to the path on the host that we would normally map into the container.

Reply:
Double checked the YAMLs and seems like the UDS paths on the host can be directly passed as [parameters](https://github.com/kubernetes-csi/node-driver-registrar/blob/696b8b2370ab396bf4c38d5e99eeb75402db9b46/cmd/csi-node-driver-registrar/main.go#L65-L67) for the node driver registrar and the plugin registration mechanism. So for Windows node YAMLs, the hostpath mounts will just need to be removed.

Reply:
Sure. There are two UDS paths involved for CSI node plugin pods (which comprise of the node driver registrar and the CSI node plugin plus other sidecars if the plugin author configures):
1. The plugin registration UDS for registering with Kubelet's plugin watcher: created by node driver registrar [here](https://github.com/kubernetes-csi/node-driver-registrar/blob/696b8b2370ab396bf4c38d5e99eeb75402db9b46/cmd/csi-node-driver-registrar/node_register.go#L53).
2. The plugin UDS for various CSI API exchanges with kubelet: created by the CSI plugin (e.g. [gce-pd](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/pkg/gce-pd-csi-driver/server.go#L107) or [azure disk](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/8718ad848ddcb0c7dca2a51a1e4c89213b2a149e/pkg/csi-common/server.go#L86)).

More docs [here](https://github.com/kubernetes-csi/node-driver-registrar/blob/e621b4de1bea06a9597f3377b80a3f196ac2ab56/README.md#usage). So with hostprocess mode, all of these sockets would be directly created on the host.